apps/console/src/features/roles/pages/role-edit-updated.tsx

```diff
@@ -0,0 +1,188 @@
+ * Copyright (c) 2023, WSO2 LLC. (https://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { Grid } from "@oxygen-ui/react";
+import Button from "@oxygen-ui/react/Button";
+import { AlertInterface, AlertLevels, IdentifiableComponentInterface } from "@wso2is/core/models";
+import { addAlert } from "@wso2is/core/store";
+import { EmptyPlaceholder, TabPageLayout } from "@wso2is/react-components";
+import React, { FunctionComponent, ReactElement, useEffect, useState } from "react";
+import { useTranslation } from "react-i18next";
+import { useDispatch, useSelector } from "react-redux";
+import { RouteComponentProps } from "react-router";
+import { Dispatch } from "redux";
+import { Label } from "semantic-ui-react";
+import { getEmptyPlaceholderIllustrations } from "../../core/configs/ui";
+import { AppConstants } from "../../core/constants/app-constants";
+import { history } from "../../core/helpers/history";
+import { FeatureConfigInterface } from "../../core/models/config";
+import { AppState } from "../../core/store/index";
+import { useGetRoleById } from "../api";
+import { EditRole } from "../components/edit-role/edit-role-updated";
+import { RoleAudiences } from "../models/roles";
+
+type RoleEditPagePropsInterface = IdentifiableComponentInterface & RouteComponentProps;
+
+const RoleEditPage: FunctionComponent<RoleEditPagePropsInterface> = (
+    props: RoleEditPagePropsInterface
+): ReactElement =>  {
+
+    const {
+        [ "data-componentid" ]: componentId
+    } = props;
+
+    const { t } = useTranslation();
+    const dispatch: Dispatch = useDispatch();
+    const featureConfig: FeatureConfigInterface = useSelector((state: AppState) => state.config.ui.features);
+
+    const [ roleId, setRoleId ] = useState<string>(undefined);
+    const [ currentActiveTabIndex, setCurrentActiveTabIndex ] = useState<number>(0);
+
+    /**
+     * Get Role data from URL id
+     */
+    useEffect(() => {
+        const path: string[] = history.location.pathname.split("/");
+        const roleId: string = path[ path.length - 1 ];
+
+        setRoleId(roleId);
+    }, []);
+
+    const {
+        data: roleObject,
+        isLoading: isRoleDetailsRequestLoading,
+        error: roleDetailsRequestError,
+        mutate: mutateRoleObject
+    } = useGetRoleById(roleId);
+
+    /**
+     * The following useEffect is used to handle if any error occurs while fetching the role details.
```

Comment:
Shall we update the comment
```suggestion
     * Handle if any error occurs while fetching the role details.
```

Reply:
Fixed with https://github.com/wso2/identity-apps/pull/4290/commits/40c5244008b0c41e56b06852cec9b531cdf7b110